services/datamanager/builtin/capture_test.go

```diff
@@ -235,13 +198,126 @@ func TestDataCaptureEnabled(t *testing.T) {
+			newResource.EndPositionFunc = func(ctx context.Context, extra map[string]interface{}) (spatialmath.Pose, error) {
+				// Return a different value from the initial arm1 resource.
+				return spatialmath.NewPoseFromPoint(r3.Vector{X: 888, Y: 888, Z: 888}), nil
+			}
+			resources[resource] = newResource
+		}
+	}
+
+	err = dmsvc.Reconfigure(context.Background(), resources, resource.Config{
+		ConvertedAttributes: config,
+	})
+	test.That(t, err, test.ShouldBeNil)
+
+	passTimeCtx2, cancelPassTime2 := context.WithCancel(context.Background())
+	donePassingTime2 := passTime(passTimeCtx2, mockClock, captureInterval)
+
+	// Test that sensor data is captured from the new collector.
+	waitForCaptureFilesToExceedN(captureDir, len(getAllFileInfos(captureDir)))
+	testFilesContainSensorData(t, captureDir, func(t *testing.T, sd []*v1.SensorData) {
+		valueSet := map[float64]struct{}{}
+		for _, d := range sd {
+			valueSet[d.GetStruct().GetFields()["Number"].GetStructValue().GetFields()["Dual"].GetStructValue().GetFields()["Jmag"].GetNumberValue()] = struct{}{}
+		}
+		// Each resource's mocked capture method outputs a different value. Assert that we see data captured by the initial arm1 resource as well as the changed resource.
+		test.That(t, len(valueSet), test.ShouldEqual, 2)
+	})
+
+	cancelPassTime2()
+	<-donePassingTime2
+}
+
+// passTime repeatedly increments mc by interval until the context is canceled.
+func passTime(ctx context.Context, mc *clk.Mock, interval time.Duration) chan struct{} {
+	done := make(chan struct{})
+	go func() {
+		for {
+			select {
+			case <-ctx.Done():
+				close(done)
+				return
+			default:
+				mc.Add(interval)
+			}
+		}
+	}()
+	return done
+}
+
+// testFilesContainSensorData verifies that the files in `dir` contain sensor data, and calls `validateSensorData` on the data.
+func testFilesContainSensorData(t *testing.T, dir string, validateSensorData func(*testing.T, []*v1.SensorData)) {
+	t.Helper()
+	var sd []*v1.SensorData
+	filePaths := getAllFilePaths(dir)
+	for _, path := range filePaths {
+		d, err := datacapture.SensorDataFromFilePath(path)
+		// It's possible a file was closed (and so its extension changed) in between the points where we gathered
+		// file names and here. So if the file does not exist, check if the extension has just been changed.
+		if errors.Is(err, os.ErrNotExist) {
+			path = strings.TrimSuffix(path, filepath.Ext(path)) + datacapture.FileExt
+			d, err = datacapture.SensorDataFromFilePath(path)
+		}
+		test.That(t, err, test.ShouldBeNil)
+		sd = append(sd, d...)
+	}
+
+	test.That(t, len(sd), test.ShouldBeGreaterThan, 0)
+	for _, d := range sd {
+		test.That(t, d.GetStruct(), test.ShouldNotBeNil)
+		test.That(t, d.GetMetadata(), test.ShouldNotBeNil)
+	}
+
+	if validateSensorData != nil {
+		validateSensorData(t, sd)
+	}
+}
+
+// waitForCaptureFilesToExceedN returns once `captureDir` contains more than `n` files.
+func waitForCaptureFilesToExceedN(captureDir string, n int) {
```

Comment:
optional/super nitty - without reading the doc string, I found this function name a little ambiguous w.r.t. if n referred to bytes, reading count, or file count - what do you think about making the name NFiles or otherwise indicating that in the name?